extension/src/test/Events.test.ts

```diff
@@ -0,0 +1,49 @@
+import * as assert from "assert";
+import * as path from "path";
+import * as vscode from "vscode";
+import { xliffCache } from "../Xliff/XLIFFCache";
+
+suite("Events Tests", () => {
+  const cachedFilePath = path.join(
+    __dirname,
+    "../../",
+    "src/test/resources/XliffCacheTest.da-DK copy.xlf"
+  );
+  test("NABFunctions.onDidChangeTextDocument", async function () {
+    const expectedTargetContent = "Lionel Richie";
+    const newText = `\n          <target>${expectedTargetContent}</target>`;
+    const startPosition = new vscode.Position(8, 37);
+    // Clear cache
+    xliffCache.clear();
+    assert.ok(xliffCache.size === 0, "Cache is not empty.");
+    const textDocument = await vscode.workspace.openTextDocument(
+      cachedFilePath
+    );
+    const editor = await vscode.window.showTextDocument(textDocument);
+    editor.edit((editBuilder) => {
+      editBuilder.insert(startPosition, newText);
+    });
+    await textDocument.save();
+    const cachedXliff = xliffCache.get(cachedFilePath);
```

Comment:
How do we know if we got a hit in the cache, or if this function reads from the disk? 😅

maybe if we move the line `    assert.strictEqual(xliffCache.size, 1, "Expected 1 document in cache.");` above this line?

Reply:
Nice find! document is dirty so the functions bails out before caching :+1:  Looking into how to make it clean again.